server/protocols/webadmin/src/test/java/org/apache/james/webadmin/routes/DomainRoutesTest.java

```diff
@@ -0,0 +1,341 @@
+                .contentType(ContentType.JSON)
+            .when()
+                .post(DomainRoutes.DOMAIN + DOMAIN)
+            .then()
+                .statusCode(200);
+
+            given()
+                .accept(ContentType.JSON)
+                .contentType(ContentType.JSON)
+            .when()
+                .get(DomainRoutes.DOMAIN + DOMAIN)
+            .then()
+                .statusCode(200);
+        }
+
+        @Test
+        public void getDomainSHouldReturnNotFoundWhenTheDomainIsAbsent() {
+            given()
+                .accept(ContentType.JSON)
+                .contentType(ContentType.JSON)
+            .when()
+                .get(DomainRoutes.DOMAIN + DOMAIN)
+            .then()
+                .statusCode(404);
+        }
+
+    }
+
+    public class ExceptionHandling {
+
+        private DomainList domainList;
+        private String domain;
+
+        @Before
+        public void setUp() throws Exception {
+            domainList = mock(DomainList.class);
+            createServer(domainList);
+            domain = "domain";
+        }
+
+        @Test
+        public void deleteShouldReturnErrorOnUnknownException() throws Exception {
+            doThrow(new RuntimeException()).when(domainList).removeDomain(domain);
+
+            given()
+                .accept(ContentType.JSON)
+                .contentType(ContentType.JSON)
+            .when()
+                .delete(DomainRoutes.DOMAIN + "domain")
+            .then()
+                .statusCode(500);
+        }
+
+        @Test
+        public void postShouldReturnErrorOnUnknownException() throws Exception {
+            doThrow(new RuntimeException()).when(domainList).addDomain(domain);
+
+            given()
+                .accept(ContentType.JSON)
+                .contentType(ContentType.JSON)
+            .when()
+                .post(DomainRoutes.DOMAIN + "domain")
+            .then()
+                .statusCode(500);
+        }
+
+        @Test
+        public void getDomainShouldReturnErrorOnUnknownException() throws Exception {
+            when(domainList.containsDomain(domain)).thenThrow(new RuntimeException());
+
+            given()
+                .accept(ContentType.JSON)
+                .contentType(ContentType.JSON)
+            .when()
+                .get(DomainRoutes.DOMAIN + "domain")
+            .then()
+                .statusCode(500);
+        }
+
+        @Test
+        public void getDomainsShouldReturnErrorOnUnknownException() throws Exception {
+            when(domainList.getDomains()).thenThrow(new RuntimeException());
+
+            given()
+                .accept(ContentType.JSON)
+                .contentType(ContentType.JSON)
+            .when()
+                .get(DomainRoutes.DOMAINS)
+            .then()
+                .statusCode(500);
+        }
+
+        @Test
+        public void deleteShouldReturnOkWhenDomainListException() throws Exception {
```

Comment:
You are very strong to interpret comments such as "and ?" and "and so what ?" :)
anyway that should be 404, not 406 imho.

After some search, and rereading http://www.restapitutorial.com/lessons/httpmethods.html I found that to make delete idempotent you should allow delete to return 200 (or 204 for empty content) when deleting a non existing user.